src/track/trackmetadatataglib.cpp

```diff
@@ -785,6 +846,46 @@ void readTrackMetadataFromXiphComment(TrackMetadata* pTrackMetadata,
                 toQStringFirstNotEmpty(tag.fieldListMap()["GROUPING"]));
     }
 
+    if (tag.fieldListMap().contains("TRACKNUMBER")) {
+        const QString trackNumberText(
+                toQStringFirstNotEmpty(tag.fieldListMap()["TRACKNUMBER"]));
+        // Try to parse both the track number and the optional total tracks
+        // from this field
+        const std::pair<TrackNumbers, TrackNumbers::ParseResult> trackNumbersFromString(
+                TrackNumbers::fromString(trackNumberText));
+        QString trackTotalText;
+        if ((TrackNumbers::ParseResult::VALID == trackNumbersFromString.second) &&
+                !trackNumbersFromString.first.hasTotal()) {
```

Comment:
Nope: https://wiki.xiph.org/Field_names

I will add a comment with this reference.

On 01/02/2016 11:22 AM, Daniel Schürmann wrote:
>
> In src/track/trackmetadatataglib.cpp 
> <https://github.com/mixxxdj/mixxx/pull/835#discussion_r48683902>:
>
> > @@ -785,6 +846,46 @@ void readTrackMetadataFromXiphComment(TrackMetadata* pTrackMetadata,
> >                  toQStringFirstNotEmpty(tag.fieldListMap()["GROUPING"]));
> >      }
> >
> > +    if (tag.fieldListMap().contains("TRACKNUMBER")) {
> > +        const QString trackNumberText(
> > +                toQStringFirstNotEmpty(tag.fieldListMap()["TRACKNUMBER"]));
> > +        // Try to parse both the track number and the optional total tracks
> > +        // from this field
> > +        const std::pair<TrackNumbers, TrackNumbers::ParseResult> trackNumbersFromString(
> > +                TrackNumbers::fromString(trackNumberText));
> > +        QString trackTotalText;
> > +        if ((TrackNumbers::ParseResult::VALID == trackNumbersFromString.second) &&
> > +                !trackNumbersFromString.first.hasTotal()) {
>
> I am not sure, but It looks like If a metadata contains "TOTALTRACKS" it 
> should be the first choice.
> Probably related: http://forums.mp3tag.de/index.php?showtopic=16223
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/835/files#r48683902>.
>